recipes/mpfr/all/conanfile.py

```diff
@@ -56,6 +58,10 @@ def requirements(self):
             self.requires("mpir/3.0.0")
 
     def build_requirements(self):
+        self.tool_requires("autoconf/2.71")                # Needed for autoreconf when building from Git source
+        self.tool_requires("automake/1.16.5")              # Needed for aclocal when building from Git source
+        self.tool_requires("autoconf-archive/2021.02.19")  # Needed for additonal aclocal includes
+        self.tool_requires("libtool/2.4.7")                # Needed for libtool
```

Comment:
It's even worse. Now everybody has these build requirements even if they are useless to build mpfr :s

Moreover we know from past years that explicitly add autoconf, automake & libtool in build requirments doesn't scale (it's a complete mess of version conflict due to how conan works in build requirements graph).

Reply:
Hi @Spacelm,

You are correct. I had not realized that patching .ac or .m4 files triggers Make to implicitly invoke auto[re]conf. I have refactored my changes so that they *only* apply if the new autogen option is specified. Without this option enabled, there are no additional requirements nor build actions.

My motivation for including the autogen option and support is two-fold. In addition to meeting our requirement to be able to build from Git-based source, it also serves as a nice example of the efforts that have gone into porting the autoconf tools (autoconf, automake, libtool, m4, and autoconf-archive) to Conan 2.0 and demonstrates that they now work robustly on Windows as well as the other supported operating systems. Note that this latter statement is true even if not building from Git source due to the Makefile behavior noted above. Thanks